src/app/components/ModalDataElementSelector.js

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import React, { useState } from 'react';
+
+import {
+  Dialog,
+  DialogContainer,
+  Heading,
+  Content,
+  ButtonGroup,
+  Button,
+  Picker,
+  Divider,
+  Item
+} from '@adobe/react-spectrum';
+
+const handleOnSave = ({ dataElement, tokenize, platform, onSave }) => {
+  let newDataElement = '';
+
+  if (dataElement) {
+    if (!tokenize) {
+      newDataElement = dataElement;
+    } else if (platform === 'edge') {
+      newDataElement = `{{${dataElement}}}`;
+    } else {
+      newDataElement = `%${dataElement}%`;
+    }
+  }
+
+  onSave(newDataElement);
+};
+
+const dataElementList = (dataElements) =>
+  Object.values(dataElements).map((v) => ({
+    id: v.name,
+    name: v.name
+  }));
+
+export default ({ dataElements, platform, tokenize, onClose, onSave }) => {
+  const [dataElement, setDataElement] = useState('');
+
```

Comment:
I would suggest right here:

```
const dataElementList = useMemo(() => {
  Object.values(dateElements).map(({ name })) => ({
    id: name,
    name
  });
}, dataElements);
```

Then delete 43-47.

Reply:
@brenthosie I've changed it.